pkg/server/server_test.go

```diff
@@ -913,16 +921,25 @@ var _ = Describe("ApplicationsServer", func() {
 
 			})
 			It("logs server errors", func() {
-				ts := httptest.NewServer(httpHandler)
-				defer ts.Close()
-
 				errMsg := "there was a big problem"
-
+				fakeFetcher := &applicationv2fakes.FakeFetcher{}
 				// Pretend something went horribly wrong
-				kubeClient.GetApplicationsStub = func(c context.Context, s string) ([]wego.Application, error) {
-					return nil, errors.New(errMsg)
+				fakeFetcher.ListStub = func(c context.Context, s string) ([]models.Application, error) {
```

Comment:
:+1: Much more explicit unhappy-path simulation!

Nit: We recently found a slightly cleaner way of doing `fake` returns

```suggestion
				fakeFetcher.ListReturns([]models.Application{}, errors.New(errMsg))
```